### PR TITLE
only dispatch changed touch event to ts

### PR DESCRIPTION
**Pull request by bofeng-song**
Re: #

### Changelog

*

-------

### Continuous Integration

This pull request:

* [ ] needs automatic test cases check.
  > Manual trigger with `@cocos-robot run test cases` afterward.
* [ ] does not change any runtime related code or build configuration
  > If any reviewer thinks the CI checks are needed, please uncheck this option, then close and reopen the issue.

-------

### Compatibility Check

This pull request:

* [ ] changes public API, and have ensured backward compatibility with [deprecated features](https://github.com/cocos/cocos-engine/blob/v3.5.0/docs/contribution/deprecated-features.md).
* [ ] affects platform compatibility, e.g. system version, browser version, platform sdk version, platform toolchain, language version, hardware compatibility etc.
* [ ] affects file structure of the build package or build configuration which requires user project upgrade.
* [ ] **introduces breaking changes**, please list all changes, affected features and the scope of violation.

### CODE REVIEW

File: native/cocos/platform/android/AndroidPlatform.cpp [hunk 1]
```diff
@@ -222,10 +233,16 @@ class GameInputProxy {
...
             for (int i = 0; i < motionEvent->pointerCount; i++) {
+                if (index >= 0) {
+                    i = index;
+                }
                 int id = motionEvent->pointers[i].id;
                 float x = GameActivityPointerAxes_getX(&motionEvent->pointers[i]);
                 float y = GameActivityPointerAxes_getY(&motionEvent->pointers[i]);
                 touchEvent.touches.emplace_back(x, y, id);
+                if (index >= 0) {
+                    break;
```
Comment (pandamicro) [inline native/cocos/platform/android/AndroidPlatform.cpp]: So for pointer down and pointer up events, only one responsible touch will be added to the event, please confirm whether it's wanted behavior.

@dumganhar @PPpro

**Comment by dumganhar** [comment]
Add reference：https://developer.android.com/reference/android/view/MotionEvent

File: native/cocos/platform/android/AndroidPlatform.cpp [hunk 2]
```diff
@@ -408,6 +421,15 @@ class GameInputProxy {
         return _isActive;
     }
 
+private:
```
Comment (minggo) [inline native/cocos/platform/android/AndroidPlatform.cpp:424]: There is already a private section.

File: native/cocos/platform/android/AndroidPlatform.cpp [hunk 3]
```diff
@@ -206,13 +206,24 @@ class GameInputProxy {
         if (motionEvent->pointerCount > 0) {
             int action = motionEvent->action;
             int actionMasked = action & AMOTION_EVENT_ACTION_MASK;
+            int index = -1;
```
Comment (minggo) [inline native/cocos/platform/android/AndroidPlatform.cpp]: Please add a more meaningful name.

File: native/cocos/platform/android/AndroidPlatform.cpp [hunk 4]
```diff
@@ -409,6 +422,14 @@ class GameInputProxy {
     }
 
 private:
+
+    void addTouchEvent(int index, GameActivityMotionEvent *motionEvent) {
+        int id = motionEvent->pointers[index].id;
```
Comment (dumganhar) [inline native/cocos/platform/android/AndroidPlatform.cpp:430]: Whether need to check `index >= 0 && index < motionEvent->pointerCount` ?

**Review by minggo — APPROVED** [review]

**Review by dumganhar — APPROVED** [review]

**Pull request merged**